rust/cymbal/src/main.rs

```diff
@@ -113,6 +115,35 @@ async fn main() -> Result<(), Error> {
             }
         };
 
+        let store = match BasicStore::new(&config) {
+            Ok(r) => r,
+            Err(_err) => {
+                metrics::counter!(ERRORS, "cause" => "invalid_store").increment(1);
+                continue;
+            }
+        };
+
+        let resolver = ResolverImpl::new(Box::new(store));
+
+        let mut resolved_frames = Vec::new();
+        for frame in stack_trace {
+            // Do the switch case here once we determine frame type
+            let RawFrame::JavaScript(raw) = raw;
+
+            let resolved = raw.resolve(resolver, team_id);
```

Comment:
This is close to e.g. the way serde does things, with a `Deserialize` type being one that can be deserialised, and the trait implementing a function that takes a `Deserializer`, and I think makes quite a bit of sense... but passing the `ResolverImpl` directly, rather than some trait object, is a misstep I think.

Maybe the right approach here is to only have a `FrameStore` trait, that implemented `get(raw_frame_id: String) -> Option<Frame>`, and `set(id: String, frame: Frame)`, and pass that into the `RawFrame::resolve` function. The notion of a "resolver" is a bit messy anyway - the frame itself should know it's custom resolving logic - and then just having `Stores` for symbol sets and frame resolution results would be nice.

The other piece here is getting away from `SymbolSetRef` being an enum, somehow, and that's a tricky nut to crack - I can ideas, but no great ones. I'll be switching focus to error tracking tomorrow (away from CDP stuff), aside from support ofc, and will take a crack at it.

Reply:
I didn't fully follow what you had in mind for the `FrameStore` and how it might be different / interact with the existing `SymbolStore` nesting pattern. Worked on a version that removes the `ResolverImpl` in favour of passing the store around. It's probably still not perfect but might be worth working on it sync if you have some time to work it into the shape you'd imagined?

Reply:
Let's sync up tomorrow, the current CDP support queue is tamed enough that I can take an eye off it for a few hours, so tomorrow is gonna be an error tracking day (finally)